rest-api/offline/public_metrics_export.py

```diff
@@ -0,0 +1,145 @@
+from sqlalchemy import text
+from dao import database_factory
+from dao.database_utils import replace_years_old
+from participant_enums import EnrollmentStatus, OrderStatus, PhysicalMeasurementsStatus
+from participant_enums import Race, QuestionnaireStatus
+
+# TODO(calbach): consider whether we need to filter down to consented
+# individuals only.
+
+def _questionnaire_metric(name, col):
+  """Returns a metrics SQL aggregation tuple for the given key/column."""
+  return (
+      name,
+      """
+      SELECT {}, SUM(1)
+      FROM participant_summary
+      GROUP BY 1;
+      """.format(col),
+      lambda v: QuestionnaireStatus.lookup_by_number(v).name
+  )
+
+# Metrics definitions. 3-tuples of:
+# - (str) aggregation key name
+# - (str) SQL statement to select value, count for a metric (in that order)
+# - (func(str): str) optional function which takes the value from the above SQL
+#   output and converts it for presentation
+_SQL_AGGREGATIONS = [
+  ('enrollmentStatus',
+   """
+   SELECT enrollment_status, SUM(1)
+   FROM participant_summary
+   GROUP BY 1;
+   """,
+   lambda v: EnrollmentStatus.lookup_by_number(v).name),
+  # TODO(calbach): Verify whether we need to be conditionally trimming these
+  # prefixes or leaving them unmodified. Unclear if all codes will have prefix
+  # "PMI_".
+  ('gender',
+   """
+   SELECT code.value, ps.count
+   FROM (
+    SELECT gender_identity_id, SUM(1) count
+    FROM participant_summary
+    WHERE gender_identity_id IS NOT NULL
+    GROUP BY 1
+   ) ps LEFT JOIN code
+   ON ps.gender_identity_id = code.code_id;
+   """,
+   None),
+  ('race',
+   """
+   SELECT race, SUM(1)
```

Comment:
As noted previously: we don't have a simple way today of generating stuff like our Race enum values from the CDR. Perhaps that should change. 

We could alternatively (as you indicated) produce a data warehouse like this one at the point of exporting the OMOP for the CDR version. If we're going to do that, we need to be fairly sure that no participants in the CDR subsequently get filtered out or have these field values change in the process of merging with EHR data.